src/data_access/FirebaseUserDataAccessObject.java

```diff
@@ -90,13 +121,12 @@ private User getUserData(String idToken, String password) {
         throw new RuntimeException("User lookup failed, please try again.");
       }
 
-      System.out.println("User lookup successful: " + responseData);
       JSONObject responseObject = new JSONObject(responseData);
       JSONObject userObject = responseObject.getJSONArray("users").getJSONObject(0);
 
       String uid = userObject.optString("localId");
       String email = userObject.optString("email");
-      String displayName = userObject.optString("displayName", "No display name");
+      String displayName = getDisplayName(uid);
```

Comment:
Unfortunately we can't do a single Firebase query because the user's `uid` is stored at the `identitytoolkit.googleapis.com` url whereas the user's `displayName` needs to be stored at the `[PROJECT_ID].firebaseio.com` url.